File: python/bob/bob.py
"""Functions that determine response corresponding to message."""

def response(hey_bob):
    hey_bob = hey_bob.strip()
    if not hey_bob:
        return 'Fine. Be that way!'
    elif hey_bob[-1] == '?':
        if hey_bob.isupper():
            return 'Calm down, I know what I\'m doing!'
        else:
            return 'Sure.'
    elif hey_bob.isupper():
        return 'Whoa, chill out!'
    else:
        return 'Whatever.'

File: javascript/bob/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

const isUpper = (str) => {
  return str.toUpperCase() === str;
}

export const hey = (message) => {
  message = message.trim()
  if (/^[\\t\\n\\r\\s ].*$/.test(message) || message === "") {
    return 'Fine. Be that way!';
  } else if (/[a-z0-9].*\?$/.test(message) || !/[A-Za-z0-9].*/.test(message)) {
      return 'Sure.';
  }  else if (/[A-Z].*\?$/.test(message)) {
    return 'Calm down, I know what I\'m doing!';
  } else if (/[A-Z].*$/.test(message) && isUpper(message)) {
    return 'Whoa, chill out!';
  } else {
    return 'Whatever.';
  }
};
